Plugin/CatalogGraphQl/Products/Query/Search.php

```diff
@@ -57,33 +59,45 @@ class Search
     /** @var SessionService */
     private $sessionService;
 
+    /** @var CmpHelperData */
+    private $cmpHelperData;
+
     /**
      * Search constructor.
      * @param SearchResultFactory $searchResultFactory
      * @param SessionService $sessionService
+     * @param CmpHelperData $cmpHelperData
      */
     public function __construct(
         SearchResultFactory $searchResultFactory,
-        SessionService $sessionService
+        SessionService $sessionService,
+        CmpHelperData $cmpHelperData
     ) {
         $this->searchResultFactory =  $searchResultFactory;
         $this->sessionService = $sessionService;
+        $this->cmpHelperData = $cmpHelperData;
     }
 
     /**
      * @param MagentoSearch $search
      * @param array $args
      * @param ResolveInfo $info
+     * @param ContextInterface $context
+     * @return array
      * @noinspection PhpUnusedParameterInspection
      */
     // phpcs:ignore Generic.CodeAnalysis.UnusedFunctionParameter
-    public function beforeGetResult(MagentoSearch $search, array $args, ResolveInfo $info)
+    public function beforeGetResult(MagentoSearch $search, array $args, ResolveInfo $info, ContextInterface $context)
     {
         if (isset($args[self::SORT_KEY]) && isset($args[self::SORT_KEY][CategorySorting::NOSTO_PERSONALIZED_KEY])) {
             $pageSize = $args[self::PAGE_SIZE_KEY];
             $currentPage = $args[self::CURRENT_PAGE_KEY];
             $this->sessionService->setGraphqlModel(new GraphQlParamModel($pageSize, $currentPage));
+
+            $sorting = $this->cmpHelperData->getFallbackSorting();
+            $args['sort'][$sorting] = 'ASC';
```

Comment:
This will set the fallback sorting also when CM is returning results

Reply:
The fallback should be applied when the graphql query does not return any result

Reply:
I think it adds a sorting rule, it does not override it. Let me quickly re-test it...

Reply:
@olsi-qose yeah I tested, it works as a fallback it does not overwrite it.